### PR TITLE
Rename ChainReader to ContractReader

### PR DESCRIPTION
**Pull request by ilija42**
[core PR](https://github.com/smartcontractkit/chainlink/pull/14383)
[common PR](https://github.com/smartcontractkit/chainlink-common/pull/759)
[solana PR](https://github.com/smartcontractkit/chainlink-solana/pull/848)
[starknet PR](https://github.com/smartcontractkit/chainlink-starknet/pull/520)
[cosmos PR](https://github.com/smartcontractkit/chainlink-cosmos/pull/418)

### CODE REVIEW

**Review by silaslenihan — APPROVED** [review]

**Review by Farber98 — APPROVED** [review]
LGTM

File: median/plugin.go [hunk 1]
```diff
@@ -46,7 +46,8 @@ func (p *Plugin) NewMedianFactory(ctx context.Context, provider types.MedianProv
 		GasPriceSubunitsDataSource:           gasPriceSubunits,
 		IncludeGasPriceSubunitsInObservation: includeGasPriceSubunitsInObservation,
 		Logger: logger.NewOCRWrapper(lggr, true, func(msg string) {
-			ctx, cancelFn := p.stop.NewCtx()
+			var cancelFn context.CancelFunc
+			ctx, cancelFn = p.stop.NewCtx()
```
Comment (jmank88) [inline median/plugin.go:50]: We need to use a different name here instead of reassigning the `ctx` from the surrounding scope.

**Review by jmank88 — CHANGES_REQUESTED** [review]

File: median/plugin.go (hunk 1 above)
Comment (ilija42) [reply median/plugin.go:50]: idk what to do here, this same ctx seems to be used below, but it always throws shadowed declaration

Comment (ilija42) [reply median/plugin.go:50]: if I use a diff name it can't be used outside of this scope

Comment (ilija42) [reply median/plugin.go:50]: although it doesn't seem like this wrapped ctx should live outside of this scope

**Review by jmank88 — APPROVED** [review]

**Pull request merged**